app/services/packages_fetcher.ts

```diff
@@ -93,6 +110,33 @@ export class PackagesFetcher {
       packages = packages.filter((pkg) => pkg.category === options.category)
     }
 
+    if (options.parties) {
+      packages = packages.filter((pkg) => {
+        return options.parties!.includes(pkg.type)
+      })
+    }
+
+    if (options.versions) {
+      const specifiersRegex = /[\^~]/g
```

Comment:
I think we can extract this into a `filterVersionPackages` method in the same class

Reply:
We can do it but don't we have to do the same for the `party` and `seach` option?

Reply:
Yeah, but the logic for filtering by version is rather low-level and verbose. Having a : 

```ts
    if (options.category) {
      packages = packages.filter((pkg) => pkg.category === options.category)
    }
```

doesn't particularly shock me, for example, it's just a single line of code, without much logic. But for consistency's sake, we could also do it, yes, that would be cool

In any case, don't worry too much about it. Right now I have to work on something else, but I'll come back to it this weekend or tonight and I'll be able to take care of it